Packs/GoogleApigee/Integrations/GoogleApigeeEventCollector/GoogleApigeeEventCollector.py

```diff
@@ -0,0 +1,310 @@
+        ):
+            return access_token
+        # There's no token or it is expired
+        access_token, token_expiration_seconds, refresh_token = self.get_token_request(refresh_token)
+        integration_context = {
+            'access_token': access_token,
+            'token_expiration_seconds': token_expiration_seconds,
+            'token_initiate_time': time.time(),
+            'refresh_token': refresh_token,
+        }
+        demisto.info('successfully updated access token')
+        set_integration_context(context=integration_context)
+
+        return access_token
+
+    def get_token_request(self, refresh_token: str = '') -> tuple[str, str, str]:
+        """
+        Sends request to retrieve token.
+
+       Returns:
+           tuple[str, str]: token and its expiration date
+        """
+        grant_type = 'refresh_token' if refresh_token else 'password'
+        data = {
+            'username': self.username,
+            'password': self.password,
+            'grant_type': grant_type
+        }
+        if refresh_token:
+            data['refresh_token'] = refresh_token
+        headers = {
+            'Content-Type': 'application/x-www-form-urlencoded;charset=utf-8',
+            'Accept': 'application/json;charset=utf-8',
+            'Authorization': 'Basic ZWRnZWNsaTplZGdlY2xpc2VjcmV0',
+        }
+        zone = f'{self.zone}.' if self.zone else ''
+        url = f'https://{zone}login.apigee.com/oauth/token'
+        token_response = self._http_request('POST', full_url=url, url_suffix='/oauth/token', data=data, headers=headers)
+        return token_response.get('access_token'), token_response.get('expires_in'), token_response.get('refresh_token')
+
+    @staticmethod
+    def is_token_valid(token_initiate_time: float, token_expiration_seconds: float) -> bool:
+        """
+        Check whether a token has expired. A token is considered expired if it reached its expiration date in
+        seconds minus a minute.
+
+        for example ---> time.time() = 300, token_initiate_time = 240, token_expiration_seconds = 120
+
+        300.0001 - 240 < 120 - 60
+
+        Args:
+            token_initiate_time (float): the time in which the token was initiated in seconds.
+            token_expiration_seconds (float): the time in which the token should be expired in seconds.
+
+        Returns:
+            bool: True if token has expired, False if not.
+        """
+        return time.time() - token_initiate_time < token_expiration_seconds - 60
+
+    def get_logs(self, from_date, to_time: int) -> Any:
+        res = self.http_request(
+            method='GET',
+            url_suffix=f'/v1/audits/organizations/{self.org_name}',
+            params={'startTime': from_date, 'endTime': to_time, 'expand': True}
+        )
+        return res
+
+
+def search_events(client, last_run: Dict[str, float], limit: int = 0) -> tuple[List[Dict[str, Any]], Dict[str, float]]:
```

Comment:
Please split this function into smaller helper function to handle the limit, the dedup, and creating the values to return.